### PR TITLE
Feat/common kafka

### LINKED ISSUE #2592: create `@amplication/kafka` a common kafka functionality library
Gather the Kafka functionality and create a common shared library:

:upport the following use-cases and update them to use `@amplication/kafka`:
* `@amplication/git-webhook-service` - Producer (emit - publish events)
* `@amplication/git-pull-service` - Consumer (consume - process events)
* `@amplication/git-pull-request-service` - Consumer (consume- process commands and response with event)
* `@amplication/server` - Producer (send - publish commands and expect response)

### PR DESCRIPTION
**Pull request by tupe12334**
fix #2592 

## PR Details

This pr take all the logic of setup the Kafka from the env to an external lib

## PR Checklist
- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-git-pull-service/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "docker:build": "docker build ."
   },
   "dependencies": {
+    "@amplication/kafka": "^1.0.0",
```
Comment (mshidlov) [inline packages/amplication-git-pull-service/package.json:26]: please make sure that this is the same version as the global version 0.13.0

Comment (mshidlov) [inline packages/amplication-git-pull-service/package.json:26]: please make sure that this is the same version as the global version 0.13.0

File: packages/amplication-kafka/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    "prepare": "npm run build",
+    "test": "jest"
+  },
+  "keywords": [],
+  "author": "tupe12334",
+  "license": "ISC",
+  "devDependencies": {
+    "@nestjs/microservices": "8.4.4",
+    "@types/jest": "27.4.1",
+    "@types/node": "17.0.26",
```
Comment (mshidlov) [inline packages/amplication-kafka/package.json:18]: we are not using nod 17

File: packages/amplication-kafka/src/createNestjsKafkaConfig.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { KafkaOptions, Transport } from "@nestjs/microservices";
+import { ConsumerConfig } from "@nestjs/microservices/external/kafka.interface";
+import { KafkaEnvironmentVariables } from "./";
+
+export function createNestjsKafkaConfig(envSuffix: string = ""): KafkaOptions {
+  const kafkaEnv = new KafkaEnvironmentVariables(envSuffix);
+  const groupId = kafkaEnv.getGroupId();
+  let consumer: ConsumerConfig | undefined;
+  if (groupId) {
+    consumer = { ...consumer, groupId };
```
Comment (mshidlov) [inline packages/amplication-kafka/src/createNestjsKafkaConfig.ts:10]:
```suggestion
    consumer = { groupId };
```

File: packages/amplication-kafka/src/utils/environmentVariables.spec.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  it("should return a string", () => {
+    process.env[STRING_ENV_KEY] = STATIC_STRING;
+    const value = EnvironmentVariables.get(STRING_ENV_KEY, true);
+    expect(value).toBe(STATIC_STRING);
+  });
+  it("should return an array", () => {
+    process.env[ARRAY_ENV_KEY] = `["localhost:9092","localhost:3000"]`;
+    const array = EnvironmentVariables.getJson(ARRAY_ENV_KEY, true);
+    expect(Array.isArray(array)).toBe(true);
```
Comment (mshidlov) [inline packages/amplication-kafka/src/utils/environmentVariables.spec.ts:30]:
```suggestion
    expect(Array.isArray(array)).toBe(true);
    expect(array.find((data)=>data==="localhost:9092")).toBe("localhost:9092");
    expect(array.find((data)=>data==="localhost:3000")).toBe("localhost:3000");
```

File: packages/amplication-kafka/src/utils/environmentVariables.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { config } from "dotenv";
+import assert from "assert";
+
+config();
```
Comment (mshidlov) [inline packages/amplication-kafka/src/utils/environmentVariables.ts:4]: EnvironmentVariables can become singleton and config() can be part of the private contractor

File: packages/amplication-kafka/src/utils/kafkaEnv.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { EnvironmentVariables } from "./";
+import { KAFKA_BROKERS, KAFKA_CLIENT_ID, KAFKA_GROUP_ID } from "./constants";
+
+export class KafkaEnvironmentVariables extends EnvironmentVariables {
```
Comment (mshidlov) [inline packages/amplication-kafka/src/utils/kafkaEnv.ts:4]: why do you need the inheritance? you don't inherit anything all the functions are static and you don't have any properties on EnvironmentVariables

**Review by mshidlov — COMMENTED** [review]
I'm missing the webhook service update

File: packages/amplication-kafka/src/utils/environmentVariables.ts (hunk 5 above)
Comment (tupe12334) [reply packages/amplication-kafka/src/utils/environmentVariables.ts:4]: Can you extend more?

Comment (abrl91) [reply packages/amplication-kafka/src/utils/environmentVariables.ts:4]: watching 👀

Comment (mshidlov) [reply packages/amplication-kafka/src/utils/environmentVariables.ts]: @abrl91 we discussed it offline and @tupe12334  changed it to be a singleton and moved the `config()` inside the singleton constructor. after further talking the `dotenv` `config()` is not supposed to be part of this implementation at all. loading the `.env` file should happen one time globally and this should not be kafka library concern from where the environments arrived.

**Pull request merged**